Safety/Src/safety_control.c

```diff
@@ -0,0 +1,56 @@
+#include "safety_control.h"
+#include "debug.h"
+
+static Interchip_StoA_Packet* dataTX;
+static Interchip_AtoS_Packet* dataRX;
+static int16_t PPM[PPM_NUM_CHANNELS];
+
+uint16_t Safety_isManual(int16_t PWM_Value);
+
+void Safety_Init(){
+
+    PWM_Init();
+
+    dataTX = malloc(sizeof(Interchip_StoA_Packet));
+    dataRX = malloc(sizeof(Interchip_AtoS_Packet));
+    Interchip_Init(dataTX, dataRX);
+
+}
+
+void Safety_Run(){
+    while(1){
+
+        volatile int16_t *PPM_input = PPM_Get();
+        for(uint8_t i=0; i<PPM_NUM_CHANNELS; i++){
+            PPM[i] = PPM_input[i];
+        }
+
+
+        uint16_t isManual = Safety_isManual(PPM[SAFETY_CHANNEL]);
+
+        //send to Autopilot
+        for(uint8_t i=0; i<PPM_NUM_CHANNELS;i++){
+            dataTX->PWM[i] = PPM[i];
+        }
+        dataTX->safety_level = isManual;
+
+        //if not manual, change values from input PPM to Autopilot PWM values
+        if(!isManual){
+            uint8_t i;
+            uint16_t offset =1;
+            for(i=0; i<PPM_NUM_CHANNELS;i++){
+                if( (dataRX->autonomous_level & offset) !=0){
+                    PPM[i] = dataRX->PWM[i];
+                    if(PPM[i]<PWM_MIN) PPM[i] = PWM_MIN;
+                    if(PPM[i]>PWM_MAX) PPM[i] = PWM_MAX;
+                }
+                offset = offset << 1;
+            }
+        }
+        PWM_SetAll(PPM);
+    }
+}
+
+uint16_t Safety_isManual(int16_t PWM_Value){
```

Comment:
Why is this a uint16? Should be of type `bool`. Include `<stdbool.h>` if it's not already.

Reply:
It is due to the size needed for the SPI struct... Prevents me from having to cast it from bool to uint16 later on in the code, but could be argued either way. Your call.

Reply:
Cast should happen, either explicitly or implicitly, when inserting data into the structure. You're already doing implicit casts from bool (&&) to uint16_t.